### PR TITLE
Tool version reporting

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-io/bitrise/blob/master/.github/CONTRIBUTING.md)
- [ ] `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Context

Telemetry for how tool version managers are used, what are the active versions during the build, etc.

### Changes

- Define an interface for tool version reporting
- Implement this reporting logic for ASDF, the version manager that we use on the stacks. We could extend or replace this impl if needed

### CODE REVIEW

**Comment by ofalvai** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #902** <a href="https://app.graphite.dev/github/pr/bitrise-io/bitrise/902?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/bitrise-io/bitrise/902?utm_source=stack-comment).

**Review by pgyula — APPROVED** [review]

File: toolversions/toolversions.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	if code != 0 {
+		r.logger.Debugf("run asdf current: nonzero exit code: %d", code)
+		return false
+	}
+
+	return true
+}
+
+func (r *ASDFVersionReporter) CurrentToolVersions() (map[string]ToolVersion, error) {
+	cmd := r.cmdFactory.Create("asdf", []string{"current"}, &command.Opts{})
```
Comment (bitrise-silver) [inline toolversions/toolversions.go:68]: what happens to default tools (e.g. Flutter) that aren't set by `asdf` but available in our stack? Do we capture them ?

Comment (ofalvai) [reply toolversions/toolversions.go:68]: Not yet, but we are constantly trying to move more and more tools under ASDF (or any other version manager in the future).
But @godrei had a good analysis of Flutter versions recently that you might find interesting: https://bitrise.atlassian.net/wiki/spaces/ENGI/pages/2734293197/RFC+Flutter+SDK+version+on+stacks

**Comment by ofalvai** [comment]
### Merge activity

* **Jan 8, 6:14 AM**: @ofalvai started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/bitrise-io/bitrise/902).

**Pull request merged**